packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -3,21 +3,164 @@ import {
+
+  const namespaceMaxLength =
+    STACK_NAME_LENGTH_LIMIT - AMPLIFY_PREFIX.length - backendId.type.length - name.length - NUM_DASHES - HASH_LENGTH;
+
+  const namespace = sanitizeChars(backendId.namespace).slice(0, namespaceMaxLength - 1);
+
+  return ['amplify', namespace, name, backendId.type, hash].join('-');
+}
+
+const getHash = (backendId: BackendIdentifier): string =>
+  backendId.hash ??
+  // md5 would be sufficient here because this hash does not need to be cryptographically secure, but this ensures that we don't get unnecessarily flagged by some security scanner
+  createHash('sha512').update(backendId.namespace).update(backendId.name).digest('hex').slice(0, HASH_LENGTH);
+
+/**
+ * Remove all non-alphanumeric characters from the input string
+ */
+const sanitizeChars = (str: string): string => {
+  return str.replace(/[^A-Za-z0-9]/g, '');
+};
+
+export async function copyFunctionFile(projRoot: string, gen1FunctionName: string): Promise<void> {
+  const sourcePath = path.join(
+    projRoot,
+    '.amplify',
+    'migration',
+    'amplify',
+    'backend',
+    'function',
+    gen1FunctionName.split('-')[0],
+    'src',
+    'index.js',
+  );
+  const destinationPath = path.join(projRoot, 'amplify', 'function', gen1FunctionName.split('-')[0], 'handler.ts');
+  const content = await fs.readFile(sourcePath, 'utf8');
+
+  // Replace the first occurrence of 'event' with 'event: any'
+  const modifiedContent = content.replace(/(exports\.handler\s*=\s*async\s*\(\s*)event(\s*\))/, '$1event: any$2');
+
+  await fs.writeFile(destinationPath, modifiedContent, 'utf8');
+}
+
+export async function copyGen1Schema(projRoot: string): Promise<void> {
+  const gen1SchemaPath = path.join(projRoot, '.amplify', 'migration', 'amplify', 'backend', 'api', 'codegentest', 'schema.graphql');
+  const gen1Schema = await fs.readFile(gen1SchemaPath, 'utf-8');
+  const dataResourcePath = path.join(projRoot, 'amplify', 'data', 'resource.ts');
+  const dataResourceContent = await fs.readFile(dataResourcePath, 'utf-8');
+  const backendPath = path.join(projRoot, 'amplify', 'backend.ts');
+  let backendContent = await fs.readFile(backendPath, 'utf-8');
+
+  const schemaRegex = /"TODO: Add your existing graphql schema here"/;
+  const updatedContent = dataResourceContent.replace(schemaRegex, `\`${gen1Schema.trim()}\``);
+
+  const errorRegex = /throw new Error\("TODO: Add Gen 1 GraphQL schema"\);?\s*/;
+  const finalContent = updatedContent.replace(errorRegex, '');
+
+  await fs.writeFile(dataResourcePath, finalContent, 'utf-8');
+
+  const linesToAdd = `
+  const todoTable = backend.data.resources.cfnResources.additionalCfnResources['Todo'];
+  todoTable.addOverride('Properties.sseSpecification', { sseEnabled: false });
+  `;
+
+  backendContent += linesToAdd;
+  await fs.writeFile(backendPath, backendContent, 'utf-8');
+}
+
+async function setEnableGen2MigrationFeatureFlag(projectRoot: string): Promise<void> {
```

Comment:
why we need a feature flag? Is it temporary?

Reply:
We need to set this feature flag to true for the data category codegen to work, to get the table mapping information.

Reply:
Removed this function and reused already existing function to do the same from amplify-e2e-core package.